openmdao/core/system.py

```diff
@@ -1615,6 +1616,53 @@ def _init_relevance(self, mode):
         if self._use_derivatives:
             desvars = self.get_design_vars(recurse=True, get_sizes=False, use_prom_ivc=False)
             responses = self.get_responses(recurse=True, get_sizes=False, use_prom_ivc=False)
+
+            # If you have response aliases, check for overlapping indices.
+            aliases = {key: value for key, value in responses.items() if value['path']}
+            if len(aliases) > 0:
+
+                used_idx = {}
+                discrete = self._var_allprocs_discrete
+
+                for name in aliases:
+
+                    if name in discrete['input'] or name in discrete['output']:
+                        continue
+
+                    path = responses[name]['path']
+                    if path is None:
+                        path = name
+
+                    if name not in used_idx and path not in used_idx:
+                        size = self._var_allprocs_abs2meta['output'][path]['global_size']
+                        used_idx[path] = np.zeros(size, dtype=int)
+
+                        # Source won't be in aliases, but we need its indices if a constraint is
+                        # declared without an alias.
+                        if path in responses:
+                            idx = responses[path].get('indices')
+
+                            if idx is None:
+                                used_idx[path][:] += 1
+                            else:
+                                used_idx[path][idx.flat()] += 1
+                        else:
+                            # If an alias is in responses, but the path isn't, then we need to
+                            # make sure the path is present for the relevancy calculation.
+                            responses[path] = responses[name]
+
+                    indices = responses[name].get('indices')
+                    if indices is None:
```

Comment:
It seems like with this logic that we could potentially be adding +1 for the indices twice.

Reply:
It shouldn't.  The first part under the conditional only adds the indices from any constraint that was added to the variable with no alias name (if there is one). The second part adds the indices for the alias. If they both have overlapping indices, it really is a dup.